Arriba/Arriba.Test/Model/DatabaseTests.cs

```diff
@@ -40,23 +40,23 @@ public void Database_Join()
             SelectResult result;
 
             // Get Orders where OrderedByAlias is any Alias matching "T1" in People (equals JOIN)
-            result = RunQuerySet(db,
+            JoinQuery<SelectResult> q = new JoinQuery<SelectResult>(
+                db,
                 new SelectQuery() { Where = SelectQuery.ParseWhere("OrderedByAlias=#Q1[Alias]"), TableName = "Orders", Columns = new string[] { "OrderNumber" } },
-                new SelectQuery[]
-                {
-                    new SelectQuery() { Where = SelectQuery.ParseWhere("T1"), TableName = "People" }
-                });
+                new SelectQuery() { Where = SelectQuery.ParseWhere("T1"), TableName = "People" }
+            );
 
+            result = db.Query(q);
             Assert.AreEqual("O5, O3, O2, O1", JoinResultColumn(result));
 
             // Get People where Groups contains a Group Mike is in (contains self JOIN)
-            result = RunQuerySet(db,
+            q = new JoinQuery<SelectResult>(
+                db,
                 new SelectQuery() { Where = SelectQuery.ParseWhere("Groups:#Q1[Groups]"), TableName = "People", Columns = new string[] { "Alias" } },
```

Comment:
Why is this Q1 instead of Q0?  I kind of think the JoinQueries should be a dictionary instead of an IList.  Then #Q1 is actually #<key> and key is used as a look up in the dictionary.